### PR TITLE
Add logs to client

### PR DESCRIPTION
**Pull request by dvizelman**
Add logs to the client tracing job submission

### CODE REVIEW

**Review by nanimrod — APPROVED** [review]

**Comment by dvizelman** [comment]
Client logs examples

Jesque event - JOB_PROCESS job=Job [class=CustomAction, args=[12, 2.3, false, reqs], vars=null] worker=resque:worker:dimav-e7470:3257-0:JAVA_DYNAMIC_QUEUES,highPriorityQueue,lowPriorityQueue


Jesque event - JOB_EXECUTE job=Job [class=CustomAction, args=[12, 2.3, false, reqs], vars=null] worker=resque:worker:dimav-e7470:3257-0:JAVA_DYNAMIC_QUEUES,highPriorityQueue,lowPriorityQueue


Jesque event - JOB_FAILURE job=Job [class=CustomAction, args=[12, 2.3, false, reqs], vars=null] worker=resque:worker:dimav-e7470:3793-0:JAVA_DYNAMIC_QUEUES highPriorityQueue,lowPriorityQueue
java.lang.RuntimeException: Inexpected failure
	at net.greghaines.jesque.worker.custom.actions.CustomAction.run(CustomAction.java:28)
	at net.greghaines.jesque.worker.WorkerImpl.execute(WorkerImpl.java:663)
	at net.greghaines.jesque.worker.WorkerImpl.process(WorkerImpl.java:614)
	at net.greghaines.jesque.worker.WorkerImpl.poll(WorkerImpl.java:459)
	at net.greghaines.jesque.worker.WorkerImpl.run(WorkerImpl.java:221)

File: src/main/java/net/greghaines/jesque/client/AbstractClient.java [hunk 1]
```diff
@@ -347,17 +350,27 @@ private static void doEnqueue(final Jedis jedis, final String enqueueType, final
 
     private static void doEnqueue(final Jedis jedis, final String enqueueType, final String queue, final String jobJson, final String queueKey, final String queuesKey, final boolean jobUniquenessValidation, final long future) {
         final String pushStatus;
-        final String uniquenessValidation = String.valueOf(jobUniquenessValidation);
+        LOG.info("Starting " + enqueueType + " job" + jobJson + " to the queue " + queue + " with delay " + future);
```
Comment (nanimrod) [inline src/main/java/net/greghaines/jesque/client/AbstractClient.java]: This is harder to search in logz.io... please start with something more searchable.
Why aren't you using the parameters?

File: src/main/java/net/greghaines/jesque/client/AbstractClient.java [hunk 2]
```diff
@@ -347,17 +350,27 @@ private static void doEnqueue(final Jedis jedis, final String enqueueType, final
...
-            pushStatus = (String) jedis.evalsha(pushScriptHash.get(), 2, queuesKey, queueKey, enqueueType, getCurrTime(), queue, jobJson, uniquenessValidation);
-        } else{
-            pushStatus = (String) jedis.evalsha(pushScriptHash.get(), 2, queuesKey, queueKey, enqueueType, getCurrTime(), queue, jobJson, uniquenessValidation, String.valueOf(future));
+            if(future==0) {
+                pushStatus = (String) jedis.evalsha(pushScriptHash.get(), 2, queuesKey, queueKey, enqueueType, getCurrTime(), queue, jobJson, uniquenessValidation);
+            } else{
+                pushStatus = (String) jedis.evalsha(pushScriptHash.get(), 2, queuesKey, queueKey, enqueueType, getCurrTime(), queue, jobJson, uniquenessValidation, String.valueOf(future));
+            }
+        } catch (Exception e) {
+            LOG.error(enqueueType + " job " + jobJson + " to the queue "+ queue + " has failed",e);
```
Comment (nanimrod) [inline src/main/java/net/greghaines/jesque/client/AbstractClient.java]: Same thing.

File: src/main/java/net/greghaines/jesque/client/AbstractClient.java [hunk 3]
```diff
@@ -347,17 +350,27 @@ private static void doEnqueue(final Jedis jedis, final String enqueueType, final
...
+            throw e;
         }
 
         if(DUPLICATED.equals(pushStatus)){
-            throw new DuplicateJobException("Duplicated job " + jobJson + " has been found");
+            final String duplicatedJobMessage = "Duplicated job " + jobJson + " has been found";
+            LOG.warn(duplicatedJobMessage);
```
Comment (nanimrod) [inline src/main/java/net/greghaines/jesque/client/AbstractClient.java]: Same

File: src/main/java/net/greghaines/jesque/client/AbstractClient.java [hunk 4]
```diff
@@ -347,17 +350,27 @@ private static void doEnqueue(final Jedis jedis, final String enqueueType, final
...
+            throw new DuplicateJobException(duplicatedJobMessage);
         }
+
+        LOG.info(enqueueType + " job" + jobJson + " to the queue "+ queue + " has finished successfully");
```
Comment (nanimrod) [inline src/main/java/net/greghaines/jesque/client/AbstractClient.java]: same

**Review by nanimrod — CHANGES_REQUESTED** [review]

**Review by nanimrod — APPROVED** [review]

**Pull request merged**